src/metabase/driver/util.clj

```diff
@@ -659,3 +660,19 @@
           password-fields (filter #(contains? #{:password :secret} (get % :type)) all-fields)]
       (into default-sensitive-fields (map (comp keyword :name) password-fields)))
     default-sensitive-fields))
+
+(defn fetch-and-incorporate-auth-provider-details
+  "Incorporates auth-provider responses with db-details.
+
+  If you have a database you need to pass the database-id as some providers will need to save the response (e.g. refresh-tokens)."
+  ([driver db-details]
+   (fetch-and-incorporate-auth-provider-details driver nil db-details))
+  ([driver database-id {:keys [auth-provider] :as db-details}]
```

Comment:
The `auth-provider` is set in admin panel for user added dw. If we disable auth provider wanting to switch back to other auth method, will this value get cleaned up? If not, I _have a feeling_ it could lead to unwanted behavior.